pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +105,218 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
+		logger.Info("machine set is currently migrating", "machine set", mapiMachineSet.GetName())
 		return ctrl.Result{}, nil
 	default:
-		logger.Info("machine AuthoritativeAPI has unexpected value", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
+		logger.Info("Unexpected value for AuthoritativeAPI", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
 		return ctrl.Result{}, nil
 	}
 }
 
-// reconcileCAPIMachineSettoMAPIMachineSet reconciles a CAPI MachineSet to a MAPI MachineSet.
-func (r *MachineSetSyncReconciler) reconcileCAPIMachineSettoMAPIMachineSet(ctx context.Context, capiMachineSet *capiv1beta1.MachineSet, mapiMachineSet *machinev1beta1.MachineSet) (ctrl.Result, error) {
+// reconcileMAPIMachineSetToCAPIMachineSet reconciles a MAPI MachineSet to a CAPI MachineSet.
+func (r *MachineSetSyncReconciler) reconcileMAPIMachineSetToCAPIMachineSet(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
+	// This function is currently a placeholder.
 	return ctrl.Result{}, nil
 }
 
-// reconcileMAPIMachineSettoCAPIMachineSet MAPI MachineSet to a CAPI MachineSet.
-func (r *MachineSetSyncReconciler) reconcileMAPIMachineSettoCAPIMachineSet(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
+// reconcileCAPIMachineSetToMAPIMachineSet reconciles a CAPI MachineSet to a
+// MAPI MachineSet.
+//
+// TODO: Platform specific implementation (currently this works only for AWS,
+// we want a switch on platform somewhere).
+// TODO: Put Gets() for Cluster + Template in helper func.
+func (r *MachineSetSyncReconciler) reconcileCAPIMachineSetToMAPIMachineSet(ctx context.Context, capiMachineSet *capiv1beta1.MachineSet, mapiMachineSet *machinev1beta1.MachineSet) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("function", "reconcileCAPIMachineSetToMAPIMachineSet")
+
+	// Get the AWSCluster
+	cluster := &awscapiv1beta1.AWSCluster{}
+	clusterNamespacedName := client.ObjectKey{
+		Namespace: capiMachineSet.GetNamespace(),
+		Name:      capiMachineSet.Spec.ClusterName,
+	}
+
+	if err := r.Get(ctx, clusterNamespacedName, cluster); err != nil {
+		logger.Error(err, "Failed to get CAPI cluster", "cluster", clusterNamespacedName.Name)
+
+		r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachineSet, corev1.ConditionFalse,
+			"FailedToGetCluster", err.Error(),
+		)
+
+		return ctrl.Result{}, fmt.Errorf("failed to get CAPI cluster: %w", err)
+	}
+
+	// Get the AWSMachineTemplate
+	template := &awscapiv1beta1.AWSMachineTemplate{}
+	templateNamespacedName := client.ObjectKey{
+		Namespace: capiMachineSet.Spec.Template.Spec.InfrastructureRef.Namespace,
+		Name:      capiMachineSet.Spec.Template.Spec.InfrastructureRef.Name,
+	}
+
+	if err := r.Get(ctx, templateNamespacedName, template); err != nil {
+		logger.Error(err, "Failed to get AWSMachineTemplate", "name", templateNamespacedName.Name)
```

Comment:
Also here since the Get accepts a client.Object, we could stay generic and say:
```suggestion
		logger.Error(err, "Failed to get InfraMachineTemplate", "name", templateNamespacedName.Name)
```
at the top of the function, when we switch over the providers, we could then plug in the providerName in the log lines, so we carry it with us all along.